### DISCUSSION TITLE
initialize a singleton service for server routes

### DISCUSSION THREAD

**Discussion by inthegreenwild** [nuxt/nuxt#26590] [Questions] [answered]
The frontend has plugins to initialize things once and provide values across the app but I want to do this on the server. I need to initialize a 3rd party SDK and use that one instance across all server api calls but I'm unsure of the best way to approach this. Do I just create a singleton in plain JS and import that into all my /api/routes? 

what I want to do: 

initialize the sdk instance somewhere once
```
mySDK = new sdkProvider({ secret: process.env.secret });
```

then, in server/api/example.js
```
export default defineEventHandler(() => {
   const data = await mySDK.call()
    return { data }
})
```

I tried looking into server plugins but I couldn't figure out how to actually provide a value to use like with client plugins.

**Comment by TheAlexLichter** [comment, marked as answer]
An example can be found in [my Sentry recipe blog post](https://www.lichter.io/articles/nuxt3-sentry-recipe/#nitro-integration-sentry-on-the-server-side)

**Reply by inthegreenwild** [reply]
Exactly what I was looking for, thanks!
